### ISSUE TITLE
When Using a GPU: HappyTextGeneration.generate_text() Error After Training/Evaluating 

### ISSUE THREAD

**Issue by EricFillion** [EricFillion/happy-transformer#220]
Reproduction: 
- Connect to a Google Colab environment with a GPU 
- pip install happytransformer==2.2.1
- Upload a text file called "text.txt"
```

from happytransformer import HappyGeneration

happy_gen = HappyGeneration()

happy_gen.train("test.txt")

output = happy_gen.generate_text("Artificial intelligence is ")`
```

Error message:  _RuntimeError: Input, output and indices must be on the current device_

This problem is unique to HappyGeneration(). I have performed similar tests with HappyWordPrediction(), HappyQuestionAnswering() and HappyTextClassification() without finding any errors. 

Quick fix: Run the following code in-between training/evaluating and using HappyTextGeneration.generate_text()
`happy_gen.model.to("cpu")
`

**Comment by ted537** [comment]
The issue doesn't occur with the other HappyTransformers because they use pipelines.

**Comment by ted537** [comment]
You probably need to use `input_ids = input_ids.to(model._device)`

**Comment by EricFillion** [comment]
Thanks Ted! I'll try that this weekend.

**Comment by EricFillion** [comment]
See #223. Published in version 2.2.2
